### ISSUE TITLE
Entry Editor: change Optional Fields tab

### ISSUE THREAD

**Issue by MartinKarim** [JabRef/jabref#4688]
![entryoption](https://user-images.githubusercontent.com/26404402/53192010-9d1e6380-360d-11e9-9722-c5d2a378229d.png)

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
|Optional information is split between tabs (e.g. _deprecated fields_,_optional fields 2_)|unify all information that is not mandatory in the _optional fields_|if the number of fields gets large, make the liest of fields scrollable or reduce text field size|
| text fields are too big (height and width) to give a good overview of information|reduce size of text fields| -|
|dropdown fields look too similar to text fields (month)|make dropdown fields more unique| -|
|ISBN has a special role;|_ISBN_ and other identifiers are migrated to _Web Resources_|-|
|-|-|other changes are found in #4686 |

**Comment by LinusDietz** [comment]
Check out BibLaTeX Manual: ftp://ftp.tu-chemnitz.de/pub/tex/macros/latex/contrib/biblatex/doc/biblatex.pdf

**Comment by Siedlerchr** [comment]
DevCall Decision: Good idea, but be aware between the differences of biblatex and bibtex, e.g. different format numbers.
Also add a tooltip what the field represents (e.g.

**Comment by ddunig2** [comment]
Is this issue taken? If not may I tackle it?

**Comment by ilippert** [comment]
In
JabRef 5.3--2020-12-28--020cc97
Linux 5.9.16-200.fc33.x86_64 amd64 
Java 15.0.1
the layout is still the old.

**Comment by ilippert** [comment]
And I would love this change request in a real version :)

**Comment by a1819644** [comment]
Hi team, not sure if someone is looking after this issue! if not may I have a look into it.

**Comment by ThiloteE** [comment]
Hey @a1819644, yes, this Issue is free for the taking. I would suggest to start with the merger of the "optional fields", "deprecated fields" and "other fields" tabs. Jabref would visually become a lot clearer and more orderly. 👍

As a reminder, this issue here is connected to a few others that all try to enhance the entry editor and rework fields and tabs.
The meta issue that mentions the other tab-related issues is here: https://github.com/JabRef/jabref/issues/4686

**Comment by a1819644** [comment]
@ThiloteE not sure whats happening but I missing the "other fields"(please see the image). Feels like someone has fixed it and Im thinking about working on the merging the "Optional fields and Deprecated fields". 

Also, cause I haven't started typing up the code, So do you mean by creating "draft pull request " where I can mention that my team is working o
![missing the otherfields](https://user-images.githubusercontent.com/88136301/163542368-3e4aef01-fdff-4cf1-a3da-f114eaef38d8.PNG)
n this issue?

**Comment by ilippert** [comment]
> Also, cause I haven't started typing up the code, So do you mean by creating "draft pull request " where I can mention that my team is working on this issue? 

You mentioned it here, so that is sufficient. Please also notify us here if you stop working on the issue. 
For a generic intro to pull requests: just to ensure you are aware of the idea, check out https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

**Comment by ThiloteE** [comment]
The "other fields" tab will show up, when you work with entry type "article".

Just enter the following into the `{} BibTeX source` tab:

```
@Article{testauthortt,
  author = {test-author},
  title  = {test-title},
}
```

**Comment by ThiloteE** [comment]
@a1819644 Anoop, are you not working on this issue? I have seen you ask if you can work on multiple issues now, but you have not opened a single pull request. Does this mean you are NOT working on this issue anymore?

**Comment by a1819644** [comment]
@ThiloteE Ohh I'm sorry for creating misunderstanding, Im working on this issue as well. Cause my group is of 4 people we decided to choose one more issue to participate in if we can. Also ISBN issue #8652 got fixed due to change in priority therefore we decide to to choose one more issue #6448. 

make sense I will make pr as soon as possible on this issue first.

**Comment by ThiloteE** [comment]
Ok, thanks for the clarification! Great then! Yes, sorry for 8652. An unfortunate coincidence that the platform had to close down, which necessitated fast and more complex reaction by JabRef maintainers.

**Comment by a1819644** [comment]
Not a problem, I completely understand!
 ISBN search was pretty broken hahhaha. ;)

**Comment by a1819644** [comment]
@ThiloteE My team team is almost done merging the fields. Currently we are on the test phase and this is how our merging fields is going to look like. Lemme know if you would like to make  any changes before me make pr!. 
![image](https://user-images.githubusercontent.com/88136301/170208677-62a91f67-0c3c-4473-be09-79927f02d6ec.png)

**Comment by ThiloteE** [comment]
What I can see from the screenshot:

For type `article` the field `Title` is supposed to be in the `required fields` tab, not in the (merged) "optional fields" tab.

**Comment by a1819644** [comment]
> What I can see from the screenshot:
> 
> For type `article` the fields `Title` and `author` are supposed to be in the `required fields` tab, not in the (merged) "optional fields" tab.



> Hey, nice to hear from you :-)
> 
> Yes, try to open a (draft) pull request early on, so that people can see you are working on the issue and so that they can see the direction the pull request is heading towards. This way, you will likely receive valuable feedback.
> 
> By the way, meanwhile, #8783 has been merged, not sure if you have seen that.
> 
>     * **Option A) argument against merging with deprecated fields tab:**
>       I was thinking a little and by now I am also not sure anymore, if it is actually a good idea to also merge with the `deprecated fields` tab. Since that tab will be hidden anyway (thanks to 8783), unless there are deprecated fields, it will not disturb and if there are deprecated fields it will be a nice notification that there are some and the user probably should move the data from deprecated fields to supported fields :)
> 
>     * **Option B) argument for also merging with deprecated fields tab:**
>       Merging will make it so that there are less tabs, which confuse users.
>       If so, it would be nice if there were some kind of visual difference between optional _fields_ and deprecated _fields_. (e.g. colour with tooltip).
> 
> 
> Feel free to choose :) Both are fine in my opinion. Option A is probably easier, because you don't have to do anything, so maybe that's better.
> 
> Of course, `Optional Fields` and `Other fields` tab should still be merged!

Sure, Haven't seen this merge yet! Will have a look tomorrow. I will choose option A for the time being but happy to make changes later like option B because I honestly prefer the option B.

**Comment by a1819644** [comment]
> What I can see from the screenshot:
> 
> For type `article` the fields `Title` and `author` are supposed to be in the `required fields` tab, not in the (merged) "optional fields" tab.

Yes, I was testing the merging for so long that forget to change the name, thanks for the heads up.  I will make sure to change it before pulling request. Also, will remove the _### articles , authors and title_ from the field.

**Comment by ThiloteE** [comment]
For anybody who wants to work on this in the future:

The last pull-request (which unfortunately is insufficient to solve this issue here) received important and relevant feedback. Any further attempts should be aware of the following:

> [These] changes are affecting several fundamental concepts of the JabRef entry editor UI [therefore] please provide some rationale for your changes, preferably in an ADR (https://devdocs.jabref.org/adr.html) discussing the pros and cons with proper constraints (can be kept short),

> Please mind that biblatex itself is distinguishing between required entry fields and optional entry fields. Yet there are still fields left, that are not used by the entry field description for an entry type (the formerly called 'other fields') and the deprecated fields, that biblatex is not using anymore. I totally agree that using multiple tabs to provide access to the optional fields, the other fields and deprecated fields is not the most intuitive and understandable way. If you decide to merge the tabs for different kinds of fields - what you can do - you still need to somehow distinguish between the different kind of fields for an entry type, so it is clear for the user, which fields are part of the biblatex standard for that field in specific (https://ctan.ebinger.cc/tex-archive/macros/latex/contrib/biblatex/doc/biblatex.pdf) and which are not. Maybe a small heading between the fields is possible or to display them in a slightly different color...

**Comment by ilippert** [comment]
I guess this has become outdated given current development trajectory
